app/models/vm_or_template.rb

```diff
@@ -123,7 +123,8 @@ class VmOrTemplate < ActiveRecord::Base
   include ReportableMixin
 
   virtual_column :active,                               :type => :boolean
-  virtual_column :archived,                             :type => :boolean,    :uses => [:host, :storage]
+  virtual_column :archived,                             :type => :boolean,
+                 :sql  => "(CASE WHEN vms.ems_id IS NULL and vms.storage_id IS NULL THEN true ELSE false END)"
```

Comment:
This comment is not really related to the PR here, but it's more of an organization thing.  I'm concerned that if the code for the archive method is changed, that we would forget to change the SQL here.  This is because we list all the virtual_column calls up front, instead of down near the methods.  I would much rather prefer we did it after the method definition, much like we do for things like `alias_method` or `private_class_method` and the like.

So, for example, I think it would be clearer if we did something like

```ruby
def archived?
  ems_id.nil? && storage_id.nil?
end
alias_method :archived, :archived?
virtual_column :archived, :type => :boolean,
  :sql  => "(CASE WHEN vms.ems_id IS NULL and vms.storage_id IS NULL THEN true ELSE false END)"
```

What do you think? @kbrock @gtanzillo